azure/ConsiderationsForServiceDesign.md

```diff
@@ -77,6 +77,65 @@ Understanding how your service is used and defining its model and interaction pa
 
 :white_check_mark: **DO** create an [OpenAPI Definition](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md) (with [autorest extensions](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md)) describing the service. The OpenAPI definition is a key element of the Azure SDK plan and is essential for documentation, usability and discoverability of services.
 
+## Use Good Names
+
+Good names for resources, properties, operations, and parameters are essential to a good developer experience.
+
+Resources are typically described by nouns. Resource and property names must be descriptive and easy to understand for end-users. Ease of understanding comes from familiarity and recognition; you should favor consistency with other Azure services, names in the product user interface, and industry standards.
+
+Names should aid developers in discovering functionality without having to constantly refer to documentation.
+Use common patterns and standard conventions to aid developers in correctly guessing common property names and meanings.
+Use verbose naming patterns and avoid abbreviations other than
+well-known acronyms in your service domain.
+
+### Recommended Naming Conventions
+
+The following are recommended naming conventions for Azure services:
+
+:heavy_check_mark: **DO** use singular nouns for resource (schema) names.
+
+:heavy_check_mark: **DO** use plural nouns for collections (for listing type or collection properties).
+
+:ballot_box_with_check: **YOU SHOULD** case all acronyms the same as a normal word.
+
+For example, `nextUrl` and not `nextURL`.
+
+:ballot_box_with_check: **YOU SHOULD** use "at" suffix in names of `date-time` values.
+
+:ballot_box_with_check: **YOU SHOULD** use "is" prefix in names of `boolean` values.
```

Comment:
This is not something I've seen in other REST APIs. I would expect this in guidelines for a more verbose programming language, not for REST APIs. Why are we making the names verbose like this?

Reply:
I think we have given this advice in past API reviews and it also included in [Graph's naming guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/graph/GuidelinesGraph.md#:~:text=SHOULD%20prefix%20Boolean%20properties%20with%20is%2C%20unless%20this%20leads%20to%20awkward%20or%20unnatural%20sounding%20names%20for%20Boolean%20properties.)

But I would be fine to change it if others find it objectionable.

Reply:
Based on different conventions, the language generators already handle this correctly if no prefix like "is" exists. I agree that it's not common in REST (rare, though I've seen it) and doesn't seem to add much value anyway. Why waste the 2 bytes per bool property?

If we did add it - and it would presumably only exist for newer services that actually conform to guidelines - now code generators have to take the effectively optional prefix into account and handle even more cases to make sure the SDK is idiomatic.

Reply:
Preference of naming for boolean could be different. Like Golang prefer it without "is" prefix, and Java might actually like the "is" prefix in getter method (but not in the variable itself).

So we might not want it to be in REST itself.

Reply:
I'll take this one out. For languages where this is a convention it is easy to add, so leaving it out of the REST API seems better.